test/contracts/polymorphism_variance_switching_custom_types.aes

```diff
@@ -0,0 +1,132 @@
+contract interface Animal =
+    entrypoint sound : () => string
+
+contract Cat : Animal =
+    entrypoint sound() = "meow"
+
+main contract Main =
+    datatype ta('a) = TA('a => unit)
```

Comment:
Would you make the names more descriptive? Don't have to be elaborate, but so it is not necessary to constantly lookup their defs